### DISCUSSION TITLE
Guide for Flet app running on Github Pages

### DISCUSSION THREAD

**Discussion by rrendon502** [flet-dev/flet#2499] [General]
Is there a way that you could generate a guide to run a flet app running on a Github Page, I've tried several times with no success at all

**Comment by 09u2h4n** [comment]
Yes it has, i tried the 0.19.0 version.
I created a github_page titled app with ```flet create github_pages``` then i built with ```flet build web github_pages -vv``` and i found web folder, i directly upload to [my github page](https://09u2h4n.github.io/) and it worked well but it took around ~30 seconds to load on mobile with pc it was ~1-2 seconds

I recommend build with ```flet build``` not with ```flet publish``` because

Static websites built with ```flet build```, compared to ```flet publish```, have faster load time as all Python dependencies are now packaged into a single archive instead of being pulled in runtime with ```micropip```. ```flet build web``` also detects native Python packages built into Pyodide, such as ```bcrypt```, ```html5lib```, ```numpy``` and many others, and installs them from Pyodide package registry.

Some tips,
* You better change the url strategy to hash
* Some python library doesn't work because of pydoide
* Flet web app can render its UI with one of these renderers: (Copied from the docs)

1. HTML renderer - uses a combination of HTML elements, CSS, Canvas elements, and SVG elements. This renderer has a smaller download size.

2. CanvasKit renderer - this renderer is fully consistent with Flutter mobile and desktop, has faster performance with higher widget density, but adds about 2MB in download size.

3. By default, Flet uses CanvasKit renderer for both desktop and mobile browsers.

**Reply by FeodorFitsner** [reply]
We need an article to the docs! :)

**Reply by FeodorFitsner** [reply]
Took 2.24 sec for me:

![image](https://github.com/flet-dev/flet/assets/5041459/ceff3241-51a1-4ce6-9f5a-df003567f553)

**Reply by thhung** [reply]
for example, in my static website, i don't use python at all. Is there a way to get rid of Pyodide? 
2nd question: does python code is compiled to javascript or not at all?

**Reply by 09u2h4n** [reply]
I think python code is implementing into pyodide instead of converting to javascript.
And also if you use flet you are already using python.

**Reply by UltronXx** [reply]
> We need an article to the docs! :)

Can someone please and please provide a detailed description on how to host Flet 0.26 on GitHub pages?
I mean, a step by step. I don't seem to be getting anything right!

**Reply by ndonkoHenri** [reply]
Here is a video tutorial: https://youtu.be/qjHgRwQHD3s

**Reply by UltronXx** [reply]
Thank you Sir!
However, I watched your video during my endeavour.
I get it, but I don't get how to do it with the structure.

![Screenshot 2025-02-21 at 10 41 09 AM](https://github.com/user-attachments/assets/ea9f1bdd-aa7b-4064-a976-8449231b841e)

The ones I have seen were the old dir structure.
I hope you get me please

**Comment by rrendon502** [comment]
Thanks for the information

**Comment by rrendon502** [comment]
Wow! I just a rookie, cant get it right,

**Comment by rrendon502** [comment]
Great!

**Comment by alegro96** [comment]
I just wanted to try to publish a web app on GitHub pages to see if it works.

I build the web app with 'flet build web' and tested it with the python http server module. Everything worked just great. I pushed the build/web folder into my github repository (https://github.com/alegro96/Test-web-App) and published it with pages. If I visit my page there is just a blank page: https://alegro96.github.io/Test-web-App/

What I expect is this: 
![image](https://github.com/flet-dev/flet/assets/135384121/a1030bee-d341-4d08-bd58-2e9418d860f0)


Do I miss any settings or did something wrong?

**Reply by liudonghua123** [reply]
You did not config the `base_url`, see also https://flet.dev/docs/cli/publish/ and https://github.com/flet-dev/flet-build-template/blob/main/%7B%7Bcookiecutter.out_dir%7D%7D/web/index.html#L18C31-L18C40.

I check the base of your [index.html](https://alegro96.github.io/Test-web-App/index.html), it was the default `/`.

**Reply by alegro96** [reply]
This works. Thank you!

**Reply by UltronXx** [reply]
Can you help me deploy to web using the current flet directory?

**Comment by UltronXx** [comment]
Have you tried deploying a Flet web app on Cloudflare using the current
flet updates?
I mean with the new directory structure? Because now it is different from
the older ones you did your videos on
